ci_tools/check_pylint_commands.py

```diff
@@ -64,44 +105,111 @@ def check_expected_pylint_disable(file, disabled, flag, messages):
+                for value, key in disabled:
+                    for v in value:
+                        if first_iteration:
+                            messages['summary'] += f"New unexpected pylint disables found in `{f}`: `{v}`"
+                            messages['annotations'].append({
+                                'path':f,
+                                'start_line':key,
+                                'end_line':key,
+                                'annotation_level':"failure",
+                                'message':f"[ERROR] New unexpected pylint disables: {v}"})
+                            first_iteration = False
+                        else:
+                            messages['summary'] += ', `' + v + '`'
+                            if key == messages['annotations'][-1]['start_line']:
+                                messages['annotations'][-1]['message'] += ', ' + v
+                            else:
+                                messages['annotations'].append({
+                                    'path':f,
+                                    'start_line':key,
+                                    'end_line':key,
+                                    'annotation_level':"failure",
+                                    'message':f"[ERROR] New unexpected pylint disables: {v}"})
+                if not first_iteration:
+                    messages['summary'] += '\n\n'
             else:
-                messages.append(f"Unexpected pylint disables found in `{f}`: {disabled_str}")
+                for value, key in disabled:
+                    for v in value:
+                        if first_iteration:
+                            messages['summary'] += f"Unexpected pylint disables found in `{f}`: `{v}`"
+                            messages['annotations'].append({
+                                'path':f,
+                                'start_line':key,
+                                'end_line':key,
+                                'annotation_level':"warning",
+                                'message':f"Unexpected pylint disables: {v}"})
+                            first_iteration = False
+                        else:
+                            messages['summary'] += ', `' + v + '`'
+                            if key == messages['annotations'][-1]['start_line']:
+                                messages['annotations'][-1]['message'] += ', ' + v
+                            else:
+                                messages['annotations'].append({
+                                    'path':f,
+                                    'start_line':key,
+                                    'end_line':key,
+                                    'annotation_level':"warning",
+                                    'message':f"Unexpected pylint disables: {v}"})
+                if not first_iteration:
+                    messages['summary'] += '\n\n'
```

Comment:
Can we reduce some of the duplication here? E.g:
```
if file_changed:
    summary_template = "New unexpected pylint disables found in `{f}`: `{v}`"
    annotation_level = "failure"
    annotation_message = "[ERROR] New unexpected pylint disables: {v}"
else:
    summary_template = "Unexpected pylint disables found in `{f}`: `{v}`"
    annotation_level = "failure"
    annotation_message = "Unexpected pylint disables: {v}"

summary = []
for value, line in disabled:
    values_on_line = ', '.join(value)
    messages['annotations'].append({
                                'path':f,
                                'start_line':line,
                                'end_line':line,
                                'annotation_level':annotation_level,
                                'message':annotation_message.format(v=values_on_line)})
    summary.append(summary_template.format(f=f, v=values_on_line)
messages['summary'] = '\n\n'.join(summary)
```

Reply:
This snippet of code is not working but it s inspiring.